### PR TITLE
Disable Oauth check if oauth disabled

### LINKED ISSUE #32367: Gitea service panics when attempting to clone/pull
### Description

upon upgrading to gitea-1.22.3 (from 1.22.2) on windows, any clone/pull via git results in a 500 error.  The WebUI "mostly" still functions but does slow down. 
Closer examination of the logs shows a panic 

> 2024/10/29 10:34:07 ...rs/common/errpage.go:26:RenderPanicErrorPage() [E] PANIC: runtime error: invalid memory address or nil pointer dereference

since this was a MINOR bump, the service was "downgraded" to 1.22.2.
gitea-1.22.3-windows-4.0-amd64.exe  and gitea-1.22.3-gogit-windows-4.0-amd64.exe were both tried and both exhibit the same issue 

### Gitea Version

1.22.3

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

https://gist.github.com/eeyrjmr/a4eb1e5e159a71fc6152210fb030f2c3

### Screenshots

_No response_

### Git Version

2.46.1

### Operating System

Windows

### How are you running Gitea?

Windows service 

### Database

MySQL/MariaDB

### PR DESCRIPTION
**Pull request by lunny**
Fix #32367

### CODE REVIEW

**Review by wolfogre — APPROVED** [review]

**Review by Zettat123 — APPROVED** [review]

File: routers/web/web.go [hunk 1]
```diff
@@ -597,16 +599,20 @@ func registerRoutes(m *web.Router) {
...
+				m.Post("/{id}/regenerate_secret", user_setting.OAuthApplicationsRegenerateSecret)
+				m.Post("", web.Bind(forms.EditOAuth2ApplicationForm{}), user_setting.OAuthApplicationsPost)
+				m.Post("/{id}/delete", user_setting.DeleteOAuth2Application)
+				m.Post("/{id}/revoke/{grantId}", user_setting.RevokeOAuth2Grant)
+			})
+			m.Combo("/applications").Get(user_setting.Applications).
+				Post(web.Bind(forms.NewAccessTokenForm{}), user_setting.ApplicationsPost)
+		}
+
 		m.Post("/applications/delete", user_setting.DeleteApplication)
```
Comment (wxiaoguang) [inline routers/web/web.go]: I do not understand, why `m.Post("/applications/delete"` is not the same as `m.Group("/applications"` below.

File: tests/integration/oauth_test.go [hunk 2]
```diff
@@ -477,3 +479,24 @@ func TestOAuthIntrospection(t *testing.T) {
 	resp = MakeRequest(t, req, http.StatusUnauthorized)
 	assert.Contains(t, resp.Body.String(), "no valid authorization")
 }
+
+func TestGitOpWithOAuthDisabled(t *testing.T) {
```
Comment (wxiaoguang) [inline tests/integration/oauth_test.go]: What does it test? It tests nothing?

Comment (lunny) [reply tests/integration/oauth_test.go]: ee7116a7fea40a8ccfd2f18e9a2de5585517f766

File: routers/web/web.go (hunk 1 above)
Comment (lunny) [reply routers/web/web.go]: Unlike org applications UI and admin applications UI, user settings' applications UI includes both access token management and oauth2 management. I have done some changes ee7116a7fea40a8ccfd2f18e9a2de5585517f766

File: tests/integration/oauth_test.go (hunk 2 above)
Comment (wxiaoguang) [reply tests/integration/oauth_test.go]: Still, what does it test? If revert all other code, the test still passes? And I guess such "basic auth" should already have been tested somewhere else?

**Review by techknowlogick — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I do not see the test TestGitOpWithOAuthDisabled does anything useful. Removed it in 3e740107942c098f4c98952d50ea6d11512bbb04

Actually the root problem is that when there is a dot in user's password, then basic auth also calls oauth2 check, then the oauth keys are not initialized, then panic. I do not see an easy test to cover this case, it needs enough time to design a specialized test for it if you would really like to test it.

**Pull request merged**